test/unit/select_nil_test.lua

```diff
@@ -0,0 +1,54 @@
+local t = require('luatest')
+local g = t.group('select_nil')
+local luatest_capture = require('luatest.capture')
+local crud = require('crud')
+
+-- We throw a warning message when checking arguments for syntax correctness.
+-- It doesn't matter if a space exists or not (but a string argument must be)
+-- and if a select will be executed or not. We will not care about errors too
+-- because for that we have other tests.
+local any_space = 'non_existent_space'
+local expected_log = "empty or nil `select` call on user space=" .. any_space
+
+g.test_warnings = function()
+    local opts = {
+        {uc = nil, opts = nil},
+        {uc = nil, opts = {}},
+        {uc = nil, opts = {fullscan = false}},
+        {uc = {}, opts = nil},
+        {uc = {}, opts = {}},
+        {uc = {}, opts = {fullscan = false}},
+    }
+
+    for _, o in ipairs(opts) do
+        local capture = luatest_capture:new()
+        capture:wrap(true, function()
+            pcall(crud.select, any_space, o.uc, o.opts)
+        end)
+
+        local captured = capture:flush()
+	    t.assert_str_contains(captured.stderr, expected_log)
+    end
+end
+
+g.test_no_warnings = function()
+    local opts = {
+        {uc = nil, opts = {fullscan = true}},
+        {uc = {}, opts = {fullscan = true}},
+        {uc = {0}, opts = nil},
+        {uc = {0}, opts = {}},
+        {uc = {0}, opts = {fullscan = false}},
```

Comment:
I would prefer to test it with a correct condition just in case. Otherwise we lean on implementation details (order of the checks) more than it is necessary for a simple test.

Reply:
Thank you, I rewrote the test into an integration test.